modules/tensor_mechanics/src/actions/LineElementAction.C

```diff
@@ -0,0 +1,827 @@
+                               FEType(Utility::string_to_enum<Order>("FIRST"),
+                                      Utility::string_to_enum<FEFamily>("LAGRANGE")),
+                               _subdomain_id_union.empty() ? nullptr : &_subdomain_id_union);
+    }
+  }
+}
+
+void
+LineElementAction::actAddAuxKernels()
+{
+  if (_current_task == "add_aux_kernel" &&
+      (_dynamic_consistent_inertia || _dynamic_nodal_rotational_inertia ||
+       _dynamic_nodal_translational_inertia) &&
+      !_truss)
+  {
+    auto params = _factory.getValidParams("NewmarkAccelAux");
+    params.applyParameters(parameters(), {"boundary"});
+    params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+
+    for (unsigned i = 0; i < 2 * _ndisp; ++i)
+    {
+      std::string aux_kernel_name = name() + "_newmark_accel_" + Moose::stringify(i);
+
+      if (i < _ndisp)
+      {
+        params.set<AuxVariableName>("variable") = _accelerations[i];
+        params.set<std::vector<VariableName>>("velocity") = {_velocities[i]};
+        params.set<std::vector<VariableName>>("displacement") = {_displacements[i]};
+
+        _problem->addAuxKernel("NewmarkAccelAux", aux_kernel_name, params);
+      }
+      else
+      {
+        params.set<AuxVariableName>("variable") = _rot_accelerations[i - _ndisp];
+        params.set<std::vector<VariableName>>("velocity") = {_rot_velocities[i - _ndisp]};
+        params.set<std::vector<VariableName>>("displacement") = {_rotations[i - _ndisp]};
+
+        _problem->addAuxKernel("NewmarkAccelAux", aux_kernel_name, params);
+      }
+    }
+
+    params = _factory.getValidParams("NewmarkVelAux");
+    params.applyParameters(parameters(), {"boundary"});
+    params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+
+    for (unsigned i = 0; i < 2 * _ndisp; ++i)
+    {
+      std::string aux_kernel_name = name() + "_newmark_vel_" + Moose::stringify(i);
+
+      if (i < _ndisp)
+      {
+        params.set<AuxVariableName>("variable") = _velocities[i];
+        params.set<std::vector<VariableName>>("acceleration") = {_accelerations[i]};
+        _problem->addAuxKernel("NewmarkVelAux", aux_kernel_name, params);
+      }
+      else
+      {
+        params.set<AuxVariableName>("variable") = _rot_velocities[i - _ndisp];
+        params.set<std::vector<VariableName>>("acceleration") = {_rot_accelerations[i - _ndisp]};
+        _problem->addAuxKernel("NewmarkVelAux", aux_kernel_name, params);
+      }
+    }
+  }
+}
+
+void
+LineElementAction::actAddNodalKernels()
+{
+  if (_current_task == "add_nodal_kernel" && !_truss)
+  {
+    // NodalTranslationalInertia and NodalRotattionalInertia currently accept only constant real
+    // numbers for eta
+    Real eta = 0.0;
+    if (_dynamic_nodal_rotational_inertia || _dynamic_nodal_translational_inertia)
+    {
+      std::istringstream ss(getParam<MaterialPropertyName>("eta"));
+      Real real_value;
+
+      if (ss >> real_value && ss.eof())
+        eta = real_value;
```

Comment:
Don't do this yourself. We've run into a ton of different ways that this fails. Use the MooseUtils version which has a lot of capability (e.g. 1e5 is an int, etc):
https://github.com/idaholab/moose/blob/devel/framework/include/utils/MooseUtils.h#L415-L422